### PR TITLE
Overridable buttons onClickListeners

### PR DESCRIPTION
**Pull request by HugoCastelani**
Direct onClickListener replaced by overridable method.

### CODE REVIEW

**Comment by HugoCastelani** [comment]
This may not seem very useful, but if someone (like me) decide to set intro as launcher activity - which wouldn't let main acitivity content appear first -, there will be an ugly wait time between closing intro activity and opening main activity. Setting fragment as non-forwardable and overriding next button click would let me handle this.

**Comment by heinrichreimer** [comment]
To use material-intro as a launching activity please refer to the [splash screen sample](https://github.com/heinrichreimer/material-intro#splash-screens) in the sample app.
Normally you don't need to start any `Activity` from the intro.

**Comment by HugoCastelani** [comment]
Actually, I think that `IntroActivity` should start before `MainActivity` because it's kind weird two activities popping almost at same time, and you probably realize that, but it's your project and it's great the way it is : ).
Forgetting about starting an activity, there's this case where I could replace an unnecessary button function by next button:

photo_2017-07-10_12-05-46

As you can see, → button (next to `EditText`) does some validations, but I could use next button to do that. To relieve the situation, I call `nextSlide()` if it passes by validation, but I still having two buttons doing one thing. 
I bet there are more situations that this could be really useful, but it's on your own.

File: library/src/main/java/com/heinrichreimersoftware/materialintro/app/IntroActivity.java [hunk 1]
```diff
@@ -486,7 +488,7 @@ public boolean goToFirstSlide() {
         return goToSlide(0);
     }
 
-    private void performButtonBackPress() {
+    public void performButtonBackPress() {
```
Comment (heinrichreimer) [inline library/src/main/java/com/heinrichreimersoftware/materialintro/app/IntroActivity.java]: This should not be public.

File: library/src/main/java/com/heinrichreimersoftware/materialintro/app/IntroActivity.java [hunk 2]
```diff
@@ -307,6 +307,8 @@ private void initViews() {
         binding.miPager.setCurrentItem(position, false);
 
         binding.miPagerIndicator.setViewPager(binding.miPager);
+        
+        originalButtonNextOnClickListener =
```
Comment (heinrichreimer) [inline library/src/main/java/com/heinrichreimersoftware/materialintro/app/IntroActivity.java]: Nice idea to make overriding the listener reversable.

**Comment by heinrichreimer** [comment]
I see your point.

**Comment by HugoCastelani** [comment]
Done. `onClickListener` reverting could happen with direct call too, I've created a get because it makes more sense to me. Feel free to request/change whatever you want.

**Pull request merged**